gwcs/api.py

```diff
@@ -185,23 +183,29 @@ def pixel_to_world(self, *pixel_arrays):
         """
         Convert pixel values to world coordinates.
         """
-        kwargs = {}
-        if not self.forward_transform.uses_quantity:
-            kwargs = {'with_units': True}
-        return self(*pixel_arrays, **kwargs)
+        if self.forward_transform.uses_quantity:
+            pixel_arrays = pixel_arrays * u.pix  # *= only for newer astropy
```

Comment:
Instead of making the assumption that the inputs are in pixel units can you use self.input_frame.unit. Also the units on each input may be different.

Reply:
Done.